src/constraint_system/ecc/scalar_mul/fixed_base/mod.rs

```diff
@@ -7,132 +7,135 @@ use crate::constraint_system::{variable::Variable, StandardComposer};
-
-    let mut multiples = vec![ExtendedPoint::default(); num_bits];
-    multiples[0] = generator;
-    for i in 1..num_bits {
-        multiples[i] = multiples[i - 1].double();
-    }
-
-    dusk_jubjub::batch_normalize(&mut multiples).collect()
-}
-
-/// Computes a Scalar multiplication with the input scalar and a chosen generator
-pub fn scalar_mul(
-    composer: &mut StandardComposer,
-    jubjub_scalar: Variable,
-    generator: ExtendedPoint,
-) -> PointScalar {
-    // XXX: we can slice off 3 bits from the top of wnaf, since F_r prime has 252 bits.
-    // XXX :We can also move to base4 and have half the number of gates since wnaf adjacent entries product is zero, we will not go over the specified amount
-    let num_bits = 256;
-
-    // compute 2^iG
-    let mut point_multiples = compute_wnaf_point_multiples(generator, num_bits);
-    point_multiples.reverse();
-
-    // Fetch the raw scalar value as bls scalar, then convert to a jubjub scalar
-    // XXX: Not very Tidy, impl From function in JubJub
-    let raw_bls_scalar = composer.variables.get(&jubjub_scalar).unwrap();
-    let raw_jubjub_scalar = JubJubScalar::from_bytes(&raw_bls_scalar.to_bytes()).unwrap();
-
-    // Convert scalar to wnaf_2(k)
-    let wnaf_entries = raw_jubjub_scalar.compute_windowed_naf(2);
-    assert_eq!(wnaf_entries.len(), num_bits);
-
-    // Initialise the accumulators
-    let mut scalar_acc: Vec<BlsScalar> = Vec::new();
-    scalar_acc.push(BlsScalar::zero());
-    let mut point_acc: Vec<AffinePoint> = Vec::new();
-    point_acc.push(AffinePoint::identity());
-
-    // Auxillary point to help with checks on the backend
-    let mut xy_alphas = Vec::new();
-
-    // Load values into accumulators based on wnaf entries
-    for (i, entry) in wnaf_entries.iter().rev().enumerate() {
-        // Based on the WNAF, we decide what scalar and point to add
-        let (scalar_to_add, point_to_add) = match entry {
+impl StandardComposer {
+    /// Computes a Scalar multiplication with the input scalar and a chosen generator
+    pub fn fixed_base_scalar_mul(
+        &mut self,
+        jubjub_scalar: Variable,
+        generator: ExtendedPoint,
+    ) -> PointScalar {
+        // XXX: we can slice off 3 bits from the top of wnaf, since F_r prime has 252 bits.
+        // XXX :We can also move to base4 and have half the number of gates since wnaf adjacent entries product is zero, we will not go over the specified amount
+        let num_bits = 256;
+
+        // compute 2^iG
+        let mut point_multiples = compute_wnaf_point_multiples(generator, num_bits);
+        point_multiples.reverse();
+
+        // Fetch the raw scalar value as bls scalar, then convert to a jubjub scalar
+        // XXX: Not very Tidy, impl From function in JubJub
+        let raw_bls_scalar = self.variables.get(&jubjub_scalar).unwrap();
+        let raw_jubjub_scalar = JubJubScalar::from_bytes(&raw_bls_scalar.to_bytes()).unwrap();
```

Comment:
We should consider in the future changing these `unwrap` to functions that returns `Result`